logic/scripts/update_token_insight.py

```diff
@@ -366,16 +396,186 @@ def fetch_wallet_balance(wallet, label, eth_threshold):
+    fetch_wallet_balance(dist_staging_address)
+    fetch_wallet_balance(partnerships_address)
+    fetch_wallet_balance(ecosystem_growth_address)
+
+    # Update circulating supply
+    update_circulating_supply()
+
+def get_wallet_balance_from_db(wallet):
+    contact = db_models.EthContact.query.filter_by(address=wallet).first()
+
+    if contact is None:
+        return 0
+
+    return contact.ogn_balance
+
+def get_ogn_stats(format_data = True):
+    total_supply = 1000000000
+
+    ogn_price = float(redis_client.get("ogn_price") or 0)
+    locked_user_count = int(redis_client.get("locked_user_count") or 0)
+
+    results = db_models.EthContact.query.filter(db_models.EthContact.address.in_((
+        foundation_reserve_address,
+        team_dist_address,
+        investor_dist_address,
+        dist_staging_address,
+        partnerships_address,
+        ecosystem_growth_address,
+    ))).all()
+
+    ogn_balances = dict([(result.address, result.ogn_balance) for result in results])
+
+    foundation_reserve_balance = ogn_balances[foundation_reserve_address]
+    team_dist_balance = ogn_balances[team_dist_address]
+    investor_dist_balance = ogn_balances[investor_dist_address]
+    dist_staging_balance = ogn_balances[dist_staging_address]
+    partnerships_balance = ogn_balances[partnerships_address]
+    ecosystem_growth_balance = ogn_balances[ecosystem_growth_address]
+
+    locked_tokens = int(
+        foundation_reserve_balance +
+        team_dist_balance +
+        investor_dist_balance +
+        dist_staging_balance +
+        partnerships_balance +
+        ecosystem_growth_balance
+    )
+
+    circulating_supply = int(total_supply - locked_tokens)
+
+    market_cap = int(circulating_supply * ogn_price)
+
+    out_data = dict([
+        ("ogn_price", ogn_price),
+        ("circulating_supply", circulating_supply),
+        ("market_cap", market_cap),
+        ("total_supply", total_supply),
+
+        ("locked_tokens", locked_tokens),
+        ("locked_user_count", locked_user_count),
+
+        ("foundation_reserve_address", foundation_reserve_address),
+        ("team_dist_address", team_dist_address),
+        ("investor_dist_address", investor_dist_address),
+        ("dist_staging_address", dist_staging_address),
+        ("partnerships_address", partnerships_address),
+        ("ecosystem_growth_address", ecosystem_growth_address),
+    ])
+
+    if format_data:
+        out_data["ogn_price"] = '${:,}'.format(ogn_price)
+        out_data["circulating_supply"] = '{:,}'.format(circulating_supply)
+        out_data["market_cap"] = '{:,}'.format(market_cap)
+        out_data["total_supply"] = '{:,}'.format(total_supply)
+        out_data["locked_tokens"] = '{:,}'.format(locked_tokens)
+        out_data["locked_user_count"] = '{:,}'.format(locked_user_count)
+
+    return out_data
+    
+def get_supply_history():
+    results = db_models.CirculatingSupply.query.order_by(db_models.CirculatingSupply.snapshot_date.asc()).all()
```

Comment:
Do we need a limit on this query? 

lso there is no index on date. Dataset is small so will be ok but in general would be preferable to either:
 - keep query as is but add an index on created_at
 - change query to order by id (that assumes rows will be inserted in order)

Reply:
Had a chat with @matthewliu yesterday. We decided to try (and store) and show data in 10mins granularity. If that's the case, we would end up with a lot of data. We are looking at around ~48k records per year approx. 

I was thinking of using spatial/time partition (on `snapshot_date` column). Since it would be easier to query time based stats. We could use any timeseries DB plugin too. Don't have a solid plan here yet and I'm open to any suggestion on this.